app/main.py

```diff
@@ -1,2 +1,100 @@
+from __future__ import annotations
+from typing import Hashable, Any
+
+
+class Node:
+    def __init__(
+            self,
+            key: Hashable,
+            value: Any,
+            hash_val: int = None
+    ) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key) if hash_val is None else hash_val
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8, load_factor: int = 2 / 3) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [[] for _ in range(self.capacity)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        hash_val = hash(key)
+        store_index = hash_val % self.capacity
+        while self.table[store_index]:
+            if self.table[store_index][0].key == key:
+                self.table[store_index][0].value = value
+                return
+            else:
+                store_index += 1
+                if store_index >= len(self.table) - 1:
+                    store_index = 0
+
+        store = self.table[store_index]
+        store.append(Node(key, value, hash_val))
+        self.size += 1
+        if self.size >= self.capacity * self.load_factor:
+            self.resize()
+
+    def resize(self) -> None:
+        self.capacity *= 2
+        old_table = self.table
+        self.table = [[] for _ in range(self.capacity)]
+        self.size = 0
+        for store in old_table:
+            if store:
+                self.__setitem__(store[0].key, store[0].value)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hash_val = hash(key)
+        store_index = hash_val % self.capacity
+        while self.table[store_index]:
+            if self.table[store_index][0].key == key:
+                return self.table[store_index][0].value
+            store_index += 1
+            if store_index >= len(self.table) - 1:
+                store_index = 0
+        raise KeyError(key)
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __delitem__(self, key: Hashable) -> None:
+        hash_val = hash(key)
+        store = self.table[hash_val % self.capacity]
+        for i, node in enumerate(store):
+            if node.key == key:
+                del store[i]
+                self.size -= 1
+                return
+        raise KeyError(key)
+
+    def clear(self) -> None:
+        self.size = 0
+        self.table = [[] for _ in range(self.capacity)]
```

Comment:
imagine having an additional method for creating a `table` instead of manually writing this 3 times

Reply:
Done